libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -336,6 +335,14 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 		make_pair("implemented", _node.isImplemented()),
 		make_pair("scope", idOrNull(_node.scope()))
 	};
+	if (m_legacy)
```

Comment:
I think there's no harm adding `kind` to legacy mode as long as `isConstructor` is kept.

Reply:
I adjusted the last commit accordingly.